### PR TITLE
initial files

### PR DESCRIPTION
**Pull request by ramoneees**
Adding functionality to authentication, authorization coming soon.

### CODE REVIEW

File: src/framework/security/authentication/http.clj [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          (assoc :token (create-token user)))
+      false)))
+
+(defn create-basic-auth-backend
+  "Creates basic-auth backend to be used by basic-auth-middleware."
+  [db]
+  (buddy-auth-backends-httpbasic/http-basic-backend
+   {:authfn (partial basic-auth db)}))
+
+(defn create-basic-auth-middleware
```
Comment (staifa) [inline src/framework/security/authentication/http.clj:29]: I would stick to `wrap*` naming convention here.

File: src/framework/security/authentication/jwt.clj [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    (jwt/sign payload private-key {:alg :hs512})))
+
+(def token-backend
+  "Backend for verifying JWT-tokens."
+  (buddy-auth-backends/jws {:secret private-key :options {:alg :hs512}}))
+
+(defn token-auth-middleware
+  "Middleware used on routes requiring token authentication."
+  [handler]
+  (buddy-auth-middleware/wrap-authentication handler token-backend))
```
Comment (staifa) [inline src/framework/security/authentication/jwt.clj:30]: NL, would be good to create `.editorconfig` or use `zprint`. The latter would be ideal, although there's always some little formatting thing that each dev disagrees with, it sets and enforces the formatting standard for everyone so there's one less issue to argue about:)

File: src/framework/security/authentication/http.clj [hunk 3]
```diff
@@ -0,0 +1,35 @@
+(ns framework.security.authentication.http
+  (:require [buddy.auth :as buddy-auth]
+            [buddy.auth.backends :as buddy-auth-backends]
+            [buddy.auth.backends.httpbasic :as buddy-auth-backends-httpbasic]
+            [buddy.auth.middleware :as buddy-auth-middleware]
+            [buddy.hashers :as buddy-hashers]
+            [framework.security.authentication.jwt :refer [create-token]]))
+
```
Comment (staifa) [inline src/framework/security/authentication/http.clj:8]: Why multiple NLs? I know it's a small thing, but we should be consitent throughout the code, one NL is widely used standard and I don't see a reson to break it.

File: src/framework/security/passwords.clj [hunk 4]
```diff
@@ -0,0 +1,15 @@
+(ns framework.security.passwords
+  (:require [buddy.hashers :as hashers]))
+
```
Comment (staifa) [inline src/framework/security/passwords.clj:3]: Multiple NLs, see above comment.

File: src/framework/security/passwords.clj [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  "encrpyts a password, set type of encryption in options"
+  ([password]
+  (hashers/derive password))
+  ([password options]
+   (hashers/derive password options)))
+
+
+(defn check
+  [password encrypted-password]
+  (hashers/check password encrypted-password))
```
Comment (staifa) [inline src/framework/security/passwords.clj:15]: NL missing, see above comment.

File: src/framework/security/authentication/http.clj (hunk 3 above)
Comment (ramoneees) [reply src/framework/security/authentication/http.clj:8]: What it means NL? Sorry, i did not got your point here

Comment (staifa) [reply src/framework/security/authentication/http.clj:8]: new line

**Pull request closed without merging**